File: SportsBettingApp-Backend/Domain/Models/BettingTicket.cs
global using Domain.Common;
using System.Text.Json.Serialization;

namespace Domain.Models
{
    public class BettingTicket : BaseEntity
    {
       
        [JsonPropertyName("ticketPairs")]
        public List<TicketPair> TicketPairs { get; set; } = new List<TicketPair>();

        [JsonPropertyName("betAmount")]
        public double BetAmount { get; set; }

        public double BetAmountFinal { get; set; }

        [JsonPropertyName("ticketPlacedTime")]
        public DateTime TicketPlacedTime { get; set; }

        [JsonPropertyName("isWinningTicket")]
        public bool IsWinningTicket { get; set; }

        public bool IsCompleted { get; set; }

        public double WinAmount { get; set; }

        public double TotalStake { get; set; }

        public double TaxAmount { get; set; }

        public double PayoutAmount { get; set; }

        public double ManipulationCost { get; set; }

        public WalletTransaction? WalletTransaction { get; set; }
    }
}

File: SportsBettingApp/src/types/BettingTicket.ts
import SelectedPair from "@/types/SelectedPair";
import { WalletTransaction } from "@/types/WalletTransaction";
export default interface BettingTicket {
    id?: number
    ticketPairs: SelectedPair [];
    betAmount: number;
    betAmountFinal: number;
    ticketPlacedTime?: string;
    isWinningTicket?: boolean;
    isCompleted?:boolean;
    winAmount: number;
    totalStake:number;
    taxAmount:number;
    payoutAmount: number;
    manipulationCost: number;
    walletTransaction?: WalletTransaction;
}